test/e2e/data/infrastructure-vsphere/cluster-template.yaml

```diff
@@ -1,66 +1,185 @@
+apiVersion: v1
```

Comment:
I believe you need to `git rm` this file in order for the gitignore to work

Reply:
I removed it now but I feel like it would be good to keep it for visibility.
test/e2e/config/vsphere-ci.yaml will have reference to a template, that is not there.
`- sourcePath: "../../../test/e2e/data/infrastructure-vsphere/cluster-template.yaml"`

Similar to CAPZ is keeping the generated test templates: https://github.com/kubernetes-sigs/cluster-api-provider-azure/tree/master/templates/test.

Reply:
yeah it seems both dev and ci config yamls have references to the cluster-template. It's not ideal but as long as the `make e2e` target is used, the file should get generated.